language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -206,9 +205,13 @@ void setClientCapabilities(TextDocumentClientCapabilities clientCapabilities) {
             LSServiceOperationContext context = new LSServiceOperationContext();
             Hover hover;
             try {
-                List<BLangPackage> modules = ReferencesUtil.compileModulesAndFindReferences(fileUri, documentManager,
-                                                                                            position.getPosition(),
-                                                                                            context, false);
+                ReferencesUtil.findReferences(fileUri, documentManager, position.getPosition(), context);
+                LSDocument lsDocument = new LSDocument(fileUri);
```

Comment:
We need to avoid creating new LSDocument here. It's already available in the context

Reply:
Needed to calculate project Root. Thus, added it.